ADD generation of antora header nav from Hugo TOML

robnester-rh opened the pull request:
This commit satisfies HACBS-2110 and allows the navigation menu for the website to be defined in the `hugo.toml` file.
Note: Any links that are not part of the Hugo website, such as links to any portion of the Antora generated site, need to be under the key `url` instead of `pageRef` as `pageRef` is solely for Hugo generated pages.

SVG graphics can be used as menu content if the svg content is entered in the `graphic` key within the menu entry in the `hugo.toml` file.

zregvart commented on antora/supplemental-ui/helpers/parseHugoMenu.js:
```diff
@@ -0,0 +1,51 @@
+    const sortedMenuData = menuData.sort((a, b) => a.weight - b.weight);
+    return sortedMenuData;
+  } catch (error) {
+    console.error("Error parsing TOML:", error);
```
Might be better not to catch the error, i.e. if any error occurs it breaks the build

zregvart commented on website/hugo.toml:
```diff
@@ -4,7 +4,7 @@ title = 'Enterprise Contract'
 publishDir = '../public'
 copyright = "Red Hat, Inc. All rights reserved."
-relativeURLs = true
+relativeURLs = false
```
We want the relative URLs, breaks local preview otherwise

robnester-rh replied:
ack. I was setting this based on https://gohugo.io/content-management/urls/#relative-urls

zregvart replied:
Works nice for the Camel website, has the side effect of enabling link checking without going over network...

zregvart commented on antora/package.json:
```diff
@@ -26,6 +26,8 @@
     "@antora/collector-extension": "^1.0.0-alpha.3",
-    "@antora/lunr-extension": "^1.0.0-alpha.8"
+    "@antora/lunr-extension": "^1.0.0-alpha.8",
+    "fs": "^0.0.1-security",
```
We don't need this

zregvart commented on antora/package.json:
```diff
@@ -26,6 +26,8 @@
-    "@antora/lunr-extension": "^1.0.0-alpha.8"
+    "@antora/lunr-extension": "^1.0.0-alpha.8",
+    "fs": "^0.0.1-security",
+    "toml": "^3.0.0"
```
Could go to devDependencies, we're not shipping this we only need it at build time

zregvart reviewed (commented):
Looks good, I like how now the "Home" link in Antora goes to the front page instead to the documentation start page. Left a few comments, nothing major.

lcarva commented on antora/supplemental-ui/helpers/parseHugoMenu.js:
```diff
@@ -0,0 +1,46 @@
+      };
+    } else {
+      return {
+        name: item.name,
```
NIT: It would be nice to not have to duplicate the whole object:

```js
const menuItem = {
        name: item.name,
        url: item.url,
        pageRef: item.pageRef,
        weight: item.weight || 0,
        parent: item.parent,
        graphic: item.graphic,
}
if (children.length) {
  menuItem["children"] = children
}
return menuItem
```

robnester-rh committed:
ADD generation of antora header nav from Hugo TOML

This commit satisfies HACBS-2110 and allows the navigation menu for the
website to be defined in the `hugo.toml` file.
Note: Any links that are not part of the Hugo website, such as links to
any portion of the Antora generated site, need to be under the key `url`
instead of `pageRef` as `pageRef` is solely for Hugo generated pages.

SVG graphics can be used as menu content if the svg content is entered
in the `graphic` key within the menu entry in the `hugo.toml` file.

Signed-off-by: Rob Nester <rnester@redhat.com>
